src/cow-react/modules/utm/hooks.ts

```diff
@@ -40,13 +44,13 @@ export function useInitializeUtm() {
       const searchParams = new URLSearchParams(search)
       const utm = getUtmParams(searchParams)
       if (utm.utmCampaign || utm.utmCampaign || utm.utmContent || utm.utmMedium || utm.utmSource) {
-        // Only overrides the UTM if the URL includes any UTM param
+        // Only overrides the UTM if the URL includes at least one UTM param
         setUtm(utm)
-      }
 
-      ALL_UTM_PARAMS.forEach((param) => searchParams.delete(param))
-
-      navigate({ pathname, search: searchParams.toString() }, { replace: true })
+        // Clear params from URL
+        ALL_UTM_PARAMS.forEach((param) => searchParams.delete(param))
+        navigate({ pathname, search: searchParams.toString() }, { replace: true })
```

Comment:
This part should be done outside the `if` to clear out empty params.
Otherwise for example `?utm_campain=` will remain in the URL

Reply:
I don't understand. `utm_campaign` is part of `ALL_UTM_PARAMS` so it should be removed.

The IF it's triggered only if at least we have one UTM parameter. So if there's none, we shouldn't trigger it. Maybe im missing the point

Reply:
Ok, you mean, empty params? why would would we do this? 

anyways, i if you want me to handle it i can do it some other way, i would prefer to not redirect always, but only if there's some params in the URL

Reply:
Done 5e71dbf868fd931229f7fee45df3ce8272244330

Reply:
This is done for example for sell and buy amounts https://github.com/cowprotocol/cowswap/blob/1df31149f10cce2e12d81e23e4f0b3e6b5f9467c/src/cow-react/modules/swap/hooks/useSetupSwapAmountsFromUrl.ts#L41